malaria-ios/ViewControllers/SetupScreenUserProfileViewController.swift

```diff
@@ -0,0 +1,79 @@
+import UIKit
+
+class SetupScreenUserProfileViewController: UIViewController {
+  
+  @IBOutlet weak var firstNameField: UITextField!
+  @IBOutlet weak var lastNameField: UITextField!
+  @IBOutlet weak var genderField: UITextField!
+  @IBOutlet weak var ageField: UITextField!
+  @IBOutlet weak var locationField: UITextField!
+  @IBOutlet weak var emailField: UITextField!
+  @IBOutlet weak var phoneField: UITextField!
+  
+  @IBOutlet weak var scrollView: UIScrollView!
+  
+  // Provided by `PagesManagerViewController`
+  var delegate: PresentsModalityDelegate!
+  
+  override func viewDidAppear(animated: Bool) {
+    super.viewDidAppear(animated)
+    
+    if User.isUserAlreadyCreated() {
+      presentNextSetupScreen()
+    }
+  }
+  
+  func presentNextSetupScreen() {
+    appDelegate.presentSetupScreen(withDelegate: delegate)
+    dismissViewControllerAnimated(true, completion: nil)
+  }
+  
+  @IBAction func nextPressed(sender: UIButton) {
+    let firstName = firstNameField.text
+    let lastName = lastNameField.text
+    let age = ageField.text
+    let phone = phoneField.text
+    let gender = genderField.text
+    let email = emailField.text
+    let location = locationField.text
+    
+    // Try to create the user
+    let userCreationResult = User.define(firstName,
+                                         lastName: lastName,
+                                         age: age,
+                                         email: email,
+                                         gender: gender,
+                                         location: location,
+                                         phone: phone)
+    
+    if let error = userCreationResult.error {
```

Comment:
Intead of returning a tuple with the user/error you might use the try catch introduced in earlier swift. It makes more sense here. What do you think?